src/components/menu/menu.spec.ts

```diff
@@ -320,6 +346,85 @@ describe('MenuComponent', () => {
         expect(component.trigger.menu._isFocused$.value).toBeFalsy();
         expect(component.subMenuTrigger.menu._isHovering$.value).toBeFalsy();
         expect(component.subMenuTrigger.menu._isFocused$.value).toBeFalsy();
+    });
+});
+
+@Component({
+    selector: 'app-menu-test',
+    template: `
+    <div class="btn-group" *ngIf="showTrigger">
+        <button
+            type="button"
+            class="btn button-secondary dropdown-toggle"
+            [uxMenuTriggerFor]="menu">
+            Actions <ux-icon name="down" class="dropdown-icon-inline"></ux-icon>
+        </button>
+    </div>
+
+    <ux-menu #menu>
+        <button type="button" uxMenuItem>
```

Comment:
There's no *ngFor here for some reason? Isn't that the point of the test?

Reply:
The main issue was that if the trigger element is destroyed the menu was not closed, its not limited to ngFor, it could also occur when using an ngIf as well, which the test uses.

Reply:
Ah okay, I thought we were testing something about using ngFor without trackBy, but I see that's a separate issue.